### PR TITLE
Addition to introduction outlining the structure of each document

### PR DESCRIPTION
**Pull request by CameronLonsdale**
Feel free to suggest wording changes

### CODE REVIEW

File: docs/jam/jam/introduction.rst [hunk 1]
```diff
@@ -10,6 +10,8 @@ Jam is a language frontend that translates to
 :ref:`Lekvar <lekvar-introduction>`, which can then be compiled or interpreted
 by the backend.
 
+Each section will feature a worded description, an EBNF definition and where possible, coded examples to demonstrate that feature in Jam syntax.
```
Comment (BenjaminSchaaf) [inline docs/jam/jam/introduction.rst]: It's not technically EBNF. More like EBNF mixed regex, favoring readability over correctness.

Comment (CameronLonsdale) [inline docs/jam/jam/introduction.rst]: Perhaps the word "metasyntax" would be more appropriate?

Comment (BenjaminSchaaf) [inline docs/jam/jam/introduction.rst]: just `a more formal syntax definition` would be fine. Metasyntax is the syntax of a metalanguage, a language used to describe a language.

File: docs/jam/jam/introduction.rst [hunk 2]
```diff
@@ -10,6 +10,7 @@ Jam is a language frontend that translates to
 :ref:`Lekvar <lekvar-introduction>`, which can then be compiled or interpreted
 by the backend.
 
+Each section will feature a worded description, a formal syntax definition and where possible, coded examples to demonstrate that feature in Jam syntax.
```
Comment (BenjaminSchaaf) [inline docs/jam/jam/introduction.rst]: Currently all documentation wraps at 80. This makes it much easier to read the source.

**Pull request merged**